src/ir/term/mod.rs

```diff
@@ -667,6 +671,67 @@ pub struct TermData {
     pub cs: Vec<Term>,
 }
 
+/// Num Term Data to avoid HashTable lookup
+pub struct NumTerm {
+    /// the operator
+    pub op: NumOp,
+    /// Term has at most 3 children
+    pub cs: SmallVec<[usize; 3]>,
+}
+
+impl Serialize for NumTerm {
```

Comment:
`SmallVec` has no `Serialize` implementation, so it has to be implemented manually

Reply:
I see. Would it be possible to write serialize and deserialize functions for `SmallVec` and then use a standard `derive` with Serde's [`with` feature](https://serde.rs/field-attrs.html#with)?

Reply:
rust doesn't allow me to implement a trait on an external struct. It might be possible to wrap `SmallVec` in a new struct, but it seems to add some other complexity

Reply:
> rust doesn't allow me to implement a trait on an external struct.

Yep, you're right!

This is exactly one of the scenarios that "with" is designed for. It doesn't
require you to implement a *trait* for the struct, instead, you implement two
stand-alone functions, stick them in a module, derive serde implementations
for *your* struct, and use the with annotation to forward serialization of
`SmallVec` to those two functions.

Reply:
sounds cool! let me have a look at this